tests/rptest/services/redpanda.py

```diff
@@ -1013,20 +1013,22 @@ def start_redpanda(self, node):
         env_preamble = " ".join(
             [f"{k}={v}" for (k, v) in self._environment.items()])
 
+        # set llvm_profile var for code coverae
+        # each node will create its own copy of the .profraw file
+        # since each node creates a redpanda broker.
+        if self.cov_enabled():
+            llvm_profile_file = f"LLVM_PROFILE_FILE=\"{RedpandaService.COVERAGE_PROFRAW_CAPTURE}\" "
```

Comment:
nit: You could remove `LLVM_PROFILE_FILE` env var and just rely on `default.profraw`

My motivation for using `LLVM_PROFILE_FILE` is entirely a personal preference. By default, LLVM writes the profile data to `default.profraw` and that should still work with our existing coverage tools.

From [LLVM Coverage Docs](https://clang.llvm.org/docs/SourceBasedCodeCoverage.html):
>When the program exits it will write a raw profile to the path specified by the LLVM_PROFILE_FILE environment variable. If that variable does not exist, the profile is written to default.profraw in the current directory of the program. If LLVM_PROFILE_FILE contains a path to a non-existent directory, the missing directory structure will be created

Reply:
that's what I initially did, but I changed it based on [the requested changes](https://github.com/redpanda-data/redpanda/pull/7995#pullrequestreview-1236608962).. I don't have a strong preference, so I can do whatever you prefer :) 

cc @dotnwat

Reply:
It is not necessary to make the change as this discussion is only about filenames. We can revisit this If we run into this issue with `LLVM_PROFILE_FILE` again.

Reply:
the filename isn't important, per se, but i think having control over the location is useful. also, many tests restart redpanda. will this override the profraw file? if so, should we add something unique to the file name so that the analysis tools can collect all of them for a single test run?

Reply:
> the filename isn't important, per se, but i think having control over the location is useful. also, many tests restart redpanda. will this override the profraw file? 

Yes the profraw file is overwritten in this scenario.

> if so, should we add something unique to the file name so that the analysis tools can collect all of them for a single test run?

The current name for our `profraw` files is `redpanda.profraw` and that should be unique enough. We could modify our tools to specifically look for the `redpanda.profraw` name then. However, to collect coverage results for a single test run would mean we need CI to invoke scripts such as `tools/gen_coverage.py` and `tools/coverage_dash.py` on every DT test. 

IIUC, currently, we run those scripts **after** CI finished **all** DT tests so this ask would likely require a separate PR. @gousteris  what do you think?

Reply:
>The current name for our profraw files is redpanda.profraw and that should be unique enough

how would this be unique enough if the profraw file is overwritten? or maybe are considering different scenarios..

Reply:
> > The current name for our profraw files is redpanda.profraw and that should be unique enough
> 
> how would this be unique enough if the profraw file is overwritten? or maybe are considering different scenarios..

I was thinking about running coverage tools after each DT test finishes. But now that you mention it, an incremental filename would also suffice. So some naming convention like `redpanda-X.profraw` where X is the test number.

Reply:
i think we are talking about different things.

i think that the same name can be used for every test. we clean out and start from scratch for each test anyway. i think this is what you were referring to.

what i was originally thinking about was a single test that restarts redpanda. for example

1. start redpanda
2. do stuff
3. shutdown some nodes
4. do stuff
5. restart some nodes

in this case, we have a single test. but each time redpanda restarts, and this is my assumption with a single fixed file name, is that coverage for the previous instance of redpanda running will be overwritten.

however, it would be nice if each instance of redpanda gave us a new profraw file so that we could collect them and get coverage for an entire test.

does the llvm profile tooling already handle this some how?

Reply:
> what i was originally thinking about was a single test that restarts redpanda. for example
> 
>     1. start redpanda
> 
>     2. do stuff
> 
>     3. shutdown some nodes
> 
>     4. do stuff
> 
>     5. restart some nodes
> 
> 
> in this case, we have a single test. but each time redpanda restarts, and this is my assumption with a single fixed file name, is that coverage for the previous instance of redpanda running will be overwritten.
> 
> however, it would be nice if each instance of redpanda gave us a new profraw file so that we could collect them and get coverage for an entire test.
> 
> does the llvm profile tooling already handle this some how?

One of the steps to create coverage results involves merging 1-or-more profraw files into a single file. See the [doc](https://clang.llvm.org/docs/SourceBasedCodeCoverage.html) for details on `llvm-profdata merge`.

> Raw profiles have to be indexed before they can be used to generate coverage reports. This is done using the “merge” tool in llvm-profdata (which can combine multiple raw profiles and index them at the same time):

The output from this tool is then used to generate coverage reports.

So, to answer your question. If we have multiple profraw files for a single test, we simply need to call `llvm-profdata merge` on all of them before continuing with the next steps.
Our tooling should already handle that but I haven't looked at that script in a while so we may need some refactoring.

TL;DR - we should be able to handle that.

Reply:
@dotnwat @NyaliaLui could we merge this as is for now, to unblock codecov tests in CI?
current state: 
* addressed https://github.com/redpanda-data/redpanda/pull/7995#pullrequestreview-1238042300
* codecov in CI works: https://buildkite.com/redpanda/vtools/builds/5119#01859809-a660-413d-b4ba-00dc60fdde37

cc @ivotron

Reply:
@gousteris @ivotron  Looks good to me but the python linter is complaining.

Reply:
thanks Nyalia :) fyi, I fixed the linting issues, waiting for ci and then I'll merge